python/ray/tests/test_storage.py

```diff
@@ -65,22 +66,33 @@ def test_get_filesystem_s3(shutdown_only):
         assert isinstance(fs, pyarrow.fs.S3FileSystem), fs
 
 
-@pytest.mark.skipif(
-    sys.platform == "win32", reason="The issue is not fixed for windows yet"
-)
 def test_escape_storage_uri_with_runtime_env(shutdown_only):
     # https://github.com/ray-project/ray/issues/41568
     # Test to make sure we can successfully start worker process
-    # when storage uri contains ? and we use runtime env.
-    with simulate_storage("s3") as s3_uri:
+    # when storage uri contains ?,& and we use runtime env and that the
+    # moto mocking actually works with the escaped uri
+    port = 5002
+    with simulate_storage("s3", port=port) as s3_uri:
         assert "?" in s3_uri
         ray.init(storage=s3_uri, runtime_env={"env_vars": {"TEST_ENV": "1"}})
 
+        recorder_uri = f"http://localhost:{port}/moto-api/recorder"
+        requests.post(f"{recorder_uri}/start-recording")
```

Comment:
I think we don't need the recording. If we can put and then get back with the same data then it means things are working.